code/modules/language/language_holder.dm

```diff
@@ -31,8 +31,11 @@
 	copy.omnitongue = src.omnitongue
 	return copy
 
-/datum/language_holder/proc/grant_language(datum/language/dt)
-	languages[dt] = TRUE
+/datum/language_holder/proc/grant_language(datum/language/dt, shadow = FALSE)
+	if(shadow)
+		shadow_languages[dt] = TRUE
+	else
+		languages[dt] = TRUE
```

Comment:
You shouldn't ever need to modify shadow languages, they are innate.

Reply:
This going anywhere ?

Reply:
It's been a while now, but i think this change made sense and i'm fairly sure i spoke to coiax in irc about it

Reply:
Okay yeah i need this to add the language to the body instead of the mind